routes/user.js

```diff
@@ -96,12 +98,50 @@ const deleteUser = (request, response) => {
         });
 };
 
+const recipesByPantry = (request, response) => {
+    const {email,} = request.params;
+    UserServices.getUserByEmail(email)
+        .then(async data => {
+            const {user_id,} = data;
+            const pantryItems = await CurrentPantryServices.getPantryItemsOfUser(user_id);
+            let baseAPIURL = `https://spoonacular-recipe-food-nutrition-v1.p.rapidapi.com/recipes/findByIngredients?number=5&ranking=1&ignorePantry=false&ingredients=`;
+            for (let i = 0; i < pantryItems.length; i++) {
+                const ingredientStr = `${pantryItems[i].weight_left} grams of ${pantryItems[i].product_name}`
+                if (i < pantryItems.length - 1) {
+                    baseAPIURL += `${ingredientStr}%2C`;
+                } else {
+                    baseAPIURL += ingredientStr;
+                };
+            };
+            const recipes = await axios({
+                method: 'get',
+                url: baseAPIURL,
+                headers: {
+                    'X-RapidAPI-Host': 'spoonacular-recipe-food-nutrition-v1.p.rapidapi.com',
+                    'X-RapidAPI-Key': '152e0cdc26msh98d504390075f2dp1eeb72jsn4ab228d3c452',
+                },
+            });
+            response.status(200).json({
+                'msg': `Successfully retrieved recipes.`,
+                pantryItems,
+                recipes: recipes.data,
+            }); 
+        })
+        .catch(e => {
+            response.status(400).json({
+                'msg': `Something went wrong`,
+                e,
+            });
+        });
+};
+
 const getUserRouter = _ => {
     const UserRouter = express.Router();
 
     UserRouter.post('/', createUser);
     UserRouter.get('/id/:id', getUserByID);
     UserRouter.get('/email/:email', getUserByEmail);
+    UserRouter.get('/recipebypantry/:email', recipesByPantry)
```

Comment:
Is there a reason you could not pass a `userId` directly in here?
It may make it easier if you do-
`UserRouter.get('/recipebypantry/:userId', recipesByPantry)` since you would not need to make an additional check to find the `userId`.

Reply:
This is a route J created, I had the same question in his PR (PR #12). But he had a good argument for it. 
This is what he said : "I decided to use the email as a param because as soon as the user logs in, Firebase returns back the user's email to our frontend. That's why I thought using the email instead of the user_id would be more practical."